File: server/app/schemas/movie.py
from pydantic import BaseModel

class Movie(BaseModel):
    id: int
    title: str
    overview: str | None = None
    poster_path: str | None = None
    backdrop_path: str | None = None
    vote_average: float | None = None
    release_date: str | None = None

class MovieFilters(BaseModel):
    minimum_score: int
    page: int
    subgenre: str | None
    keyword: str | None

class MovieResponse(BaseModel):
    movies: list[Movie]
    page: int
    total_pages: int
    filters: MovieFilters
    

File: client/src/types/movie.ts
export interface Movie {
    id: number
    title: string
    overview: string | null
    poster_path: string | null
    backdrop_path: string | null
    vote_average: number | null
    release_date: string | null
}

export interface MovieFilters{
    minimum_score: number
    page: number
    subgenre: string | null
    keyword: string | null
}

export interface MovieResponse{
    movies: Movie[]
    page: number
    total_pages: number
    filters: MovieFilters
}
    